### PR TITLE
discovery: Fix service names for processes in containers

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

discovery: Fix service name extraction from containers

Language/framework-specific service name generation does not currently
work from containers since the code does not read the file from the
containers' root.

The service name generation code already uses the fs.FS interfaces
(according to the comments, for testing) so these can be used to ensure
that filesystem access happen inside the container's root.  However, a
problem with the fs.FS interface is that it does not allow absolute
paths nor paths with .. or ./ etc to be passed to Open/Stat/etc but the
service name generation code can do this in several places.  To handle
this, we have our own fs.FS interface implementation with wraps fs.DirFS
to handle this transparently for the callers.

In order to ensure that this code is exercised in the tests, the
existing "testdata" directories are moved into "testdata/root", and the
fs.FS mechanism is used to set the latter as the root directory during
testing.

With this in place, make system-probe pass the /proc/$PID/root as the
root to get the service name generation to work for containers.

### Motivation

Pre-requisite for https://datadoghq.atlassian.net/browse/USMON-1198

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go [hunk 1]
```diff
@@ -410,6 +411,25 @@ func TestAPMInstrumentationProvided(t *testing.T) {
...
+
+	require.NoError(t, nodejs.RunServerNodeJS(t, key, cert, "4444"))
+	nodeJSPID, err := nodejs.GetNodeJSDockerPID()
+	require.NoError(t, err)
+
+	url := setupDiscoveryModule(t)
+	pid := int(nodeJSPID)
+
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		portMap := getServicesMap(t, url)
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go:426]:
```suggestion
		svcMap := getServicesMap(t, url)
```

File: pkg/collector/corechecks/servicediscovery/usm/jboss.go [hunk 2]
```diff
@@ -98,6 +98,9 @@ func (j jbossExtractor) findDeployedApps(domainHome string) ([]jeeDeployment, bo
 		log.Debug("jboss: unable to extract the home directory")
 		return nil, false
 	}
+	if cwd, ok := workingDirFromEnvs(j.cxt.envs); ok {
+		baseDir = abs(baseDir, cwd)
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/usm/jboss.go:103]: worth adding documentation

File: pkg/collector/corechecks/servicediscovery/usm/jboss_test.go [hunk 3]
```diff
@@ -227,22 +230,22 @@ func TestJbossFindDeployedApps(t *testing.T) {
 			args: []string{
 				"-Djboss.home.dir=" + jbossTestAppRoot,
 			},
-			domainHome: "" + jbossTestAppRoot + "/standalone",
+			domainHome: "/testdata/a" + "/standalone",
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/usm/jboss_test.go:233]:
```suggestion
			domainHome: "/testdata/a/standalone",
```

File: pkg/collector/corechecks/servicediscovery/usm/jboss_test.go [hunk 4]
```diff
@@ -227,22 +230,22 @@ func TestJbossFindDeployedApps(t *testing.T) {
...
 			expected: []jeeDeployment{
 				{
 					name: "app.ear",
-					path: "" + jbossTestAppRoot + "/standalone/data/content/38/e/content",
+					path: "" + jbossTestAppRootAbsolute + "/standalone/data/content/38/e/content",
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/usm/jboss_test.go:237]: why do we have `"" +`?

**Review by guyarb — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/usm/jboss.go (hunk 2 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/usm/jboss.go:103]: Added.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**